test/unit/test_tokens.c

```diff
@@ -0,0 +1,233 @@
+		{ .start = 7, .len = 2, .type = RZ_ASM_TOKEN_SEPARATOR, .val.number = 0 }, // )\s
+		{ .start = 9, .len = 2, .type = RZ_ASM_TOKEN_REGISTER, .val.number = 0 }, // R5
+		{ .start = 11, .len = 1, .type = RZ_ASM_TOKEN_SEPARATOR, .val.number = 0 }, // :
+		{ .start = 12, .len = 1, .type = RZ_ASM_TOKEN_NUMBER, .val.number = 4 }, // 4
+		{ .start = 13, .len = 1, .type = RZ_ASM_TOKEN_SEPARATOR, .val.number = 0 }, // \s
+		{ .start = 14, .len = 1, .type = RZ_ASM_TOKEN_OPERATOR, .val.number = 0 }, // =
+		{ .start = 15, .len = 1, .type = RZ_ASM_TOKEN_SEPARATOR, .val.number = 0 }, // \s
+		{ .start = 16, .len = 4, .type = RZ_ASM_TOKEN_MNEMONIC, .val.number = 0 }, // memd
+		{ .start = 20, .len = 1, .type = RZ_ASM_TOKEN_SEPARATOR, .val.number = 0 }, // (
+		{ .start = 21, .len = 2, .type = RZ_ASM_TOKEN_REGISTER, .val.number = 0 }, // R0
+		{ .start = 23, .len = 1, .type = RZ_ASM_TOKEN_OPERATOR, .val.number = 0 }, // +
+		{ .start = 24, .len = 2, .type = RZ_ASM_TOKEN_REGISTER, .val.number = 0 }, // R2
+		{ .start = 26, .len = 2, .type = RZ_ASM_TOKEN_OPERATOR, .val.number = 0 }, // <<
+		{ .start = 28, .len = 1, .type = RZ_ASM_TOKEN_SEPARATOR, .val.number = 0 }, // #
+		{ .start = 29, .len = 3, .type = RZ_ASM_TOKEN_NUMBER, .val.number = 1 }, // 0x1
+		{ .start = 32, .len = 1, .type = RZ_ASM_TOKEN_SEPARATOR, .val.number = 0 } // )
+	};
+	RzAsmTokenString *toks = rz_asm_token_string_new();
+	rz_print_tokenize_asm_string(asm_str, toks, NULL);
+
+	mu_assert_eq(rz_vector_len(toks->tokens), 20, "Number of generated tokens");
+
+	int i = 0;
+	RzAsmToken *it;
+	rz_vector_foreach(toks->tokens, it) {
+		mu_assert_eq(it->start, tokens[i].start, "Token start");
+		mu_assert_eq(it->len, tokens[i].len, "Token length");
+		mu_assert_eq(it->type, tokens[i].type, "Token type");
+		mu_assert_eq(it->val.number, tokens[i].val.number, "Token value");
+		++i;
+	}
+
+	rz_strbuf_free(asm_str);
+	mu_end;
+}
+
+static bool test_rz_tokenize_generic_2(void) {
+	RzStrBuf *asm_str = rz_strbuf_new("mov pc, 🍍");
+	RzAsmToken tokens[5] = {
+		{ .start = 0, .len = 3, .type = RZ_ASM_TOKEN_MNEMONIC, .val.number = 0 },
+		{ .start = 3, .len = 1, .type = RZ_ASM_TOKEN_SEPARATOR, .val.number = 0 },
+		{ .start = 4, .len = 2, .type = RZ_ASM_TOKEN_MNEMONIC, .val.number = 0 },
+		{ .start = 6, .len = 2, .type = RZ_ASM_TOKEN_SEPARATOR, .val.number = 0 },
+		{ .start = 8, .len = 4, .type = RZ_ASM_TOKEN_UNKNOWN, .val.number = 0 }
+	};
+	RzAsmTokenString *toks = rz_asm_token_string_new();
+	rz_print_tokenize_asm_string(asm_str, toks, NULL);
+
+	mu_assert_eq(rz_vector_len(toks->tokens), 5, "Number generated tokens");
+
+	int i = 0;
+	RzAsmToken *it;
+	rz_vector_foreach(toks->tokens, it) {
+		mu_assert_eq(it->start, tokens[i].start, "Token start");
+		mu_assert_eq(it->len, tokens[i].len, "Token length");
+		mu_assert_eq(it->type, tokens[i].type, "Token type");
+		mu_assert_eq(it->val.number, tokens[i].val.number, "Token value");
+		++i;
+	}
+
```

Comment:
what are you really testing here? I think you are testing `rz_print_tokenize_asm_string` and if so, IMHO you should test the bytes written in the rz_cons buffer, not the toks->tokens.

Reply:
In this specific case I checked for UTF-8 characters in the asm string.
These tests are not about coloring yet. So no `rz_cons` involved yet (please correct me if I'm wrong. Haven't had used `rz_cons` yet.).
I just wanted to make sure all the indices, size and types of the tokens are correct.